presto-main/src/main/java/com/facebook/presto/operator/PagesSpatialIndexSupplier.java

```diff
@@ -137,7 +144,15 @@ private static STRtree buildRTree(LongArrayList addresses, List<List<Block>> cha
                 partition = toIntExact(INTEGER.getLong(partitionBlock, blockPosition));
             }
 
-            rtree.insert(getJtsEnvelope(ogcGeometry, radius), new GeometryWithPosition(ogcGeometry, partition, position));
+            GeometryWithPosition geometryWithPosition = new GeometryWithPosition(ogcGeometry, partition, position);
+            rtree.insert(getJtsEnvelope(ogcGeometry, radius), geometryWithPosition);
+
+            estimatedLeafNodesSize = estimatedLeafNodesSize + ENVELOPE_INSTANCE_SIZE + geometryWithPosition.getEstimatedMemorySizeInBytes();
+
+            // update memory size when 1000 more leaf nodes are inserted
+            if (position % 1000 == 999) {
```

Comment:
I think it would be better to update the memory every so many bytes (or megabytes) rather than every so many objects.

Reply:
How about 100 MB?  for a cluster with 600 nodes, this means > 60GB total user memory change per update.